stable_baselines3/common/off_policy_algorithm.py

```diff
@@ -154,13 +154,23 @@ def __init__(
     def _setup_model(self) -> None:
         self._setup_lr_schedule()
         self.set_random_seed(self.seed)
-        self.replay_buffer = ReplayBuffer(
-            self.buffer_size,
-            self.observation_space,
-            self.action_space,
-            self.device,
-            optimize_memory_usage=self.optimize_memory_usage,
-        )
+
+        if isinstance(self.observation_space, gym.spaces.Dict):
```

Comment:
looks like duplicate code... probably using the if/else to choose the class and then pass the same parameters to the constructors